### ISSUE TITLE
[Feature]: Consistency for the SNOWFLAKE_AUTHENTICATOR value

### ISSUE THREAD

**Issue by vludax** [Snowflake-Labs/terraform-provider-snowflake#2983]
### Use Cases or Problem Statement

As [documented in the migration guide](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/MIGRATION_GUIDE.md#behavior-change-authenticator-jwt), since version 0.74.0 of the provider, the `authenticator` property has to be set to "JWT" to use key-pair authentication.

In order to simplify our upgrade process, we would like to make use of the `SNOWFLAKE_AUTHENTICATOR` environment variable. However, we have discovered that several other connectors, such as [python](https://docs.snowflake.cn/en/developer-guide/python-connector/python-connector-connect)  and [nodejs](https://docs.snowflake.com/en/developer-guide/node-js/nodejs-driver-authenticate) expect the value of `SNOWFLAKE_AUTHENTICATOR` to be "SNOWFLAKE_JWT" rather than "JWT. It also looks like the "SNOWFLAKE_JWT" value is also [used by the go SDK](https://github.com/snowflakedb/gosnowflake/blob/master/auth.go#L120).

### Category

category:provider_config

### Proposal

Let the terraform provider treat the "SNOWFLAKE_JWT" value the same way it treats "JWT" for the `authenticator` property.

### How much impact is this issue causing?

Medium

### Would you like to implement a fix?

- [ ] Yeah, I'll take it 😎

**Comment by sfc-gh-jcieslak** [comment]
Hey @vludax 👋 
Thanks for reporting the issue. Soon we'll be going through [provider configuration rework](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/ROADMAP.md#providers-configuration-rework) where we'll address topics such as this one. But until that happens, please continue using `JWT` for now. Any breaking changes will be documented in the [migration guide](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/MIGRATION_GUIDE.md), but we'll ping you here once the change will be released.

**Comment by sfc-gh-jmichalak** [comment]
@vludax You can also use a config like this:
```
variable "authenticator" {
  type = string

}
provider "snowflake" {
  ...
  authenticator = var.authenticator
}
```
and run terraform like `TF_VAR_authenticator="JWT" terraform refresh`.
 See https://developer.hashicorp.com/terraform/cli/config/environment-variables#tf_var_name for more details.

**Comment by vludax** [comment]
@sfc-gh-jmichalak The reason we wanted to use the `SNOWFLAKE_AUTHENTICATOR` environment variable was to avoid making additional changes in the provider config. If we need to add `authenticator = var.authenticator`, we may as well just set `authenticator = "JWT"` 🙂

**Comment by sfc-gh-jmichalak** [comment]
Hi @vludax   👋

We've released a new v0.98.0 version ([release](https://github.com/Snowflake-Labs/terraform-provider-snowflake/releases/tag/v0.98.0), [migration guide](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/MIGRATION_GUIDE.md#v0970--v0980)). For now, we accept both of these values, but `SNOWFLAKE_JWT` is the preferred method and `JWT` is deprecated and will be removed in the future releases.

**Comment by vludax** [comment]
Thank you @sfc-gh-jmichalak!
